### PR TITLE
Bump golangci-lint 1.46.2 -> 1.47.3

### PR DESCRIPTION
**Pull request by dfarrell07**
https://github.com/golangci/golangci-lint/releases/tag/v1.47.3
https://github.com/golangci/golangci-lint/compare/v1.46.2...v1.47.3

This resolves golang 1.18 limitations that we've discussed.

golangci/golangci-lint/pull/3001

Depends on https://github.com/submariner-io/shipyard/issues/900
Signed-off-by: Daniel Farrell <dfarrell@redhat.com>

### CODE REVIEW

**Comment by dfarrell07** [comment]
Interesting new failures. Raised https://github.com/submariner-io/shipyard/issues/900 to track.

**Comment by tpantelis** [comment]
> Depends on [submariner-io/subctl#199](https://github.com/submariner-io/subctl/pull/199)

Won't we have a chicken and the egg thing where linting will fail and gate https://github.com/submariner-io/subctl/pull/199 which will gate this PR but this PR is needed to unblock https://github.com/submariner-io/subctl/pull/199? I think we need to merge this PR first and fix up **subctl** et al after.

**Comment by dfarrell07** [comment]
> > Depends on [submariner-io/subctl#199](https://github.com/submariner-io/subctl/pull/199)
> 
> Won't we have a chicken and the egg thing where linting will fail and gate [submariner-io/subctl#199](https://github.com/submariner-io/subctl/pull/199) which will gate this PR but this PR is needed to unblock [submariner-io/subctl#199](https://github.com/submariner-io/subctl/pull/199)? I think we need to merge this PR first and fix up **subctl** et al after.

Yep, you're right, this will need to go in before the subctl nilerr PR. I updated the Depends-on line here to track the gosec issues, which I think we should fix before this is merged.

**Review by tpantelis — APPROVED** [review]

**Review by nyechiel — APPROVED** [review]

**Pull request merged**
